### PR TITLE
Init ipvsInterface only when ipvs modules are present

### LINKED ISSUE #61074: proxy report ipvs related warning even ipvs not enabled
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>

/kind bug

> /kind feature

**What happened**:
kube proxy report following error even if ipvs not enabled.
```
W0306 09:14:47.817778       1 server.go:185] WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.
time="2018-03-06T09:14:47Z" level=warning msg="Running modprobe ip_vs failed with message: `modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/4.4.0-116-generic/modules.dep.bin'\nmodprobe: WARNING: Module ip_vs not found in directory /lib/modules/4.4.0-116-generic`, error: exit status 1"
time="2018-03-06T09:14:47Z" level=error msg="Could not get ipvs family information from the kernel. It is possible that ipvs is not enabled in your kernel. Native loadbalancing will not work until this is fixed."
W0306 09:14:47.844870       1 server_others.go:289] Flag proxy-mode="" unknown, assuming iptables proxy
I0306 09:14:47.846223       1 server_others.go:138] Using iptables Proxier.
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Install kubernetes and start proxy.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):

We are using 1.9.1

```
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.1+icp", GitCommit:"75dbaf83789341a0782ffaab75cb97ca6fb40b16", GitTreeState:"clean", BuildDate:"2018-02-23T08:07:45Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.1+icp", GitCommit:"75dbaf83789341a0782ffaab75cb97ca6fb40b16", GitTreeState:"clean", BuildDate:"2018-02-23T08:07:45Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/cc @m1093782566 comments?

### PR DESCRIPTION
**Pull request by islinwb**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61074 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by gyliu513 — COMMENTED** [review]
/lgtm

Thanks @islinwb

**Comment by m1093782566** [comment]
@islinwb 

Can you run `kube-proxy --cleanup` to make sure we can clean up ipvs left-overs?

In addition, we should discuss the semantic of 

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L492-L494

Whether we should return `true` or `false` directly when ipvs interface is nil.

**Comment by islinwb** [comment]
@m1093782566 OK.

File: cmd/kube-proxy/app/server_others.go [hunk 1]
```diff
@@ -92,9 +92,14 @@ func newProxyServer(
 
 	dbus = utildbus.New()
 	iptInterface = utiliptables.New(execer, dbus, protocol)
-	ipvsInterface = utilipvs.New(execer)
 	kernelHandler = ipvs.NewLinuxKernelHandler()
 	ipsetInterface = utilipset.New(execer)
+	proxyMode := getProxyMode(string(config.Mode), iptInterface, kernelHandler, ipsetInterface, iptables.LinuxKernelCompatTester{})
+	if proxyMode == proxyModeIPVS {
+		ipvsInterface = utilipvs.New(execer)
+	} else {
```
Comment (mikedanese) [inline cmd/kube-proxy/app/server_others.go]: This else is unnecessary, it's already nil.

Comment (islinwb) [reply cmd/kube-proxy/app/server_others.go]: @mikedanese Exactly.

File: pkg/proxy/ipvs/proxier.go [hunk 2]
```diff
@@ -547,6 +547,9 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
 
 // CleanupLeftovers clean up all ipvs and iptables rules created by ipvs Proxier.
 func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset utilipset.Interface, cleanupIPVS bool) (encounteredError bool) {
+	if canUse, _ := CanUseIPVSProxier(NewLinuxKernelHandler(), ipset); !canUse {
```
Comment (islinwb) [inline pkg/proxy/ipvs/proxier.go]: @m1093782566 Maybe we can assume ipvs mode has not been used ealier if `canUse` is false at the moment.

Comment (m1093782566) [reply pkg/proxy/ipvs/proxier.go]: SGTM!

**Comment by m1093782566** [comment]
/lgtm

/assign @thockin 

for final review :)

**Pull request merged**